www/js/app.js

```diff
@@ -63,6 +63,11 @@ angular.module('emission', ['ionic',
             Logger.displayError("Error reading or setting connection defaults", err);
         });
     });
+
+    $rootScope.$on('$stateChangeStart', function (event, toState, toStateParams, fromState, fromParams) {
```

Comment:
do you still need this? In an earlier change, you had indicated that this was primarily for development - e.g. 

> That is only for development purpose. It appears that if I opened this page and the code got reloaded, the back button disappear. This help fix it by navigate back to diary page. It's not perfect but it works on the development environment and it should not interfere with the live deployment.

Reply:
This part is now mandatory. It helps to go back to the diary page when user save the survey.